compiler/src/iree/compiler/Dialect/VMVX/Transforms/ResolveBufferDescriptors.cpp

```diff
@@ -322,13 +443,20 @@ class ResolveBufferDescriptorsPass
   ResolveBufferDescriptorsPass() = default;
   ResolveBufferDescriptorsPass(const ResolveBufferDescriptorsPass &) {}
   void getDependentDialects(DialectRegistry &registry) const override {
-    registry.insert<IREE::VMVX::VMVXDialect>();
+    registry.insert<AffineDialect, IREE::VMVX::VMVXDialect>();
   }
 
   void runOnOperation() override {
     RewritePatternSet patterns(&getContext());
-    patterns.insert<FromAllocation, FromGlobal, FromHalInterfaceBindingSubspan,
-                    FromMemRefSubView>(&getContext());
+    patterns.insert<FromAllocation<GetBufferDescriptorOp>,
+                    FromAllocation<memref::ExtractStridedMetadataOp>,
+                    FromGlobal<GetBufferDescriptorOp>,
+                    FromGlobal<memref::ExtractStridedMetadataOp>,
+                    FromHalInterfaceBindingSubspan,
+                    FromMemRefSubView<GetBufferDescriptorOp>,
+                    FromMemRefSubView<memref::ExtractStridedMetadataOp>,
```

Comment:
FYI, it is possible to expand the `subview` into "simpler" `memref` operations by using the `expand-strided-metadata` pass (or use the related `populate` function.)
I.e., we could drop the handling of `subview`s here.